### ISSUE TITLE
[false positive] `This condition will never hold` for mutable var

### ISSUE THREAD

**Issue by jrudolph** [HairyFotr/linter#7]
```
[warn] /home/johannes/git/telfish/spray/spray/spray-io/src/main/scala/akka/io/DirectByteBufferPool.scala:43: This condition will never hold.
[warn]         try if (buffersInPool > 0) 
```

`buffersInPool` is a mutable field of the class this condition lies in.
